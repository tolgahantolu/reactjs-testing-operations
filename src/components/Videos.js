import React from "react";

const Videos = ({ videos = [] }) => {
  if (videos.length <= 0) {
    return <h1>No videos found!</h1>;
  }

  return (
    <div>
      {videos.map((video, i) => (
        <h3 key={i}>{video}</h3>
      ))}
    </div>
  );
};

export default Videos;
