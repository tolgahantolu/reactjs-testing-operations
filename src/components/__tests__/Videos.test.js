import React from "react";
import renderer from "react-test-renderer";
import Videos from "../Videos";

// !Snapshot testing kullanıcı arayüzünün beklenmedik bir şekilde değişip değişmediğini kontrol etmemizi sağlıyor.
test("videos empty renders 'No videos found!'", () => {
  const tree = renderer.create(<Videos />).toJSON(); // DOM tree

  expect(tree).toMatchSnapshot();
});

// !İlk çalıştığında snapshot olmadığı için bir snapshot alır ve kaydeder. Daha sonra bir değişiklik meydana geldiğinde ona bakarak karşılaştırma yapar. Bu değişikliği kabul etmek istersek command olarak u -update- girmemiz yeterli olur, daha sonra snaphot güncellenecektir.

test("videos 1 item renders correctly", () => {
  const tree = renderer
    .create(<Videos videos={["udemy", "youtube", "coursera", "hackerrank"]} />)
    .toJSON(); // DOM tree

  expect(tree).toMatchSnapshot();
});
