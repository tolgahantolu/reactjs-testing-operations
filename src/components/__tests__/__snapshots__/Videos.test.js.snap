// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`videos 1 item renders correctly 1`] = `
<div>
  <h3>
    udemy
  </h3>
  <h3>
    youtube
  </h3>
  <h3>
    coursera
  </h3>
  <h3>
    hackerrank
  </h3>
</div>
`;

exports[`videos empty renders 'No videos found!' 1`] = `
<h1>
  No videos found!
</h1>
`;
